tools/python_integration_tests/classes/ssh.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2024 "Google LLC"
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import subprocess
+import time
+import paramiko
+
+class SSHManager:
+    # Manages tunnel and SSH connection.
+    _instance = None
+
+    def __new__(cls, *args, **kwargs):
+       if not cls._instance:
+           cls._instance = super(SSHManager, cls).__new__(cls)
+       return cls._instance
+
+    def __init__(self, deployment):
+        if not hasattr(self, 'ssh_client'):
+            self.deployment = deployment
+            self.tunnel = None
+            self.key = None
+            self.ssh_client = self._establish_ssh_connection()
+
+    def create_tunnel(self):
+        iap_tunnel_cmd = [
+            "gcloud", "compute", "start-iap-tunnel", self.deployment.instance_name,
```

Comment:
I think that `SSH` class should be agnostic to `deployment`, escpecially if want to SSH to multiple machines witin cluster.

Reply:
Ack, to be discussed